storybookjs/storybook
# CLI: Remove deprecated `-s` flag on `sb init`

Issue: #19420 

## What I did
Previously, generators could pass a `staticDir` to the `baseGenerator` and it would then be included in the newly added `npm run storybook` script (which would be something like `storybook dev -p 6006 -s public`). However, [the `-s` flag has been deprecated](https://storybook.js.org/docs/react/api/cli-options#static-dir-deprecation) in favor of the `staticDirs` property in `.storybook/main.js`.

Therefore, I now instead pass the `staticDir` to the function that creates this config file. The config file requires an array of strings instead of just a normal string and since it's located in a subdirectory, the relative paths must all be changed. In order to preserve the `baseGenerator`'s interface, I chose to perform the necessary transformations on `staticDir`. If you would prefer to modify `baseGenerator`'s interface, just let me know and I'll be happy to do it.

## How to test
1. Clone this branch, build and link the CLI
2. Create an empty CRA app (`npx create-react-app my-react-app` and `cd my-react-app`)
3. Run `path-to-your-clone/code/lib/cli/bin/index.js init` (in order to test against the current version instead of my change, run `npx @storybook/cli init`)
4. `cat package.json`, verify that the `storybook` and `build-storybook` commands don't contain the `-s` flag
5. `cat .storybook/main.js`, verify that the `staticDirs` property is defined and contains `"../public"`
6. `npm run storybook` and verify that the public folder is in fact loaded


If your answer is yes to any of these, please make sure to include it in your PR.
